content/spin/redis-trigger.md

```diff
@@ -23,54 +30,154 @@ The Redis instance address is specified in the application trigger:
 
-The Redis trigger is built on top of the
-[WebAssembly component model](https://github.com/WebAssembly/component-model).
-The current interface is defined using the
-[WebAssembly Interface (WIT)](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md)
-format, and is a function that takes the message payload as its only parameter:
+The Redis component interface is defined using a WebAssembly Interface (WIT) file.  ([Learn more about the evolving WIT standard here.](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md)).  You can find the latest WITs for Spin Redis components at https://github.com/fermyon/spin/blob/main/wit/ephemeral.
+
+The core Redis types are defined in https://github.com/fermyon/spin/blob/main/wit/ephemeral/redis-types.wit, though note that not all of these are used in the pub-sub Redis trigger:
 
 <!-- @nocpy -->
 
 ```fsharp
-// wit/ephemeral/spin-redis.wit
+// General purpose error.
+enum error {
+    success,
+    error,
+}
 
 // The message payload.
 type payload = list<u8>
 
+// Remaining types are not used in the trigger
+```
+
+> The same Redis types are also used to model the API for sending outbound Redis requests.
+
+The entry point for Spin Redis components is then defined in https://github.com/fermyon/spin/blob/main/wit/ephemeral/spin-redis.wit:
+
+<!-- @nocpy -->
+
+```fsharp
+// wit/ephemeral/spin-redis.wit
+
+use * from redis-types
+
 // The entry point for a Redis handler.
-handle-redis-message: function(msg: payload) -> expected<_, error>
+handle-redis-message: func(message: payload) -> expected<unit, error>
 ```
 
-> The interface might change in the future to add the Redis instance and
-> message channel as arguments to the function.
+This is the function signature that all Redis components must implement, and
+which is used by Spin when instantiating and invoking the component.
 
-This is the function that all Redis components must implement, and which is
-used by the Spin Redis executor when instantiating and invoking the component.
 This interface (`spin-redis.wit`) can be directly used together with the
 [Bytecode Alliance `wit-bindgen` project](https://github.com/bytecodealliance/wit-bindgen)
-to build a component that the Spin HTTP executor can invoke.
-This is exactly how [the Rust SDK for Spin](./rust-components.md) is built, and,
-as more languages add support for the component model, how we plan to add
-support for them as well.
-
-> We are [exploring a compatibility layer (similar to Wagi)](https://github.com/fermyon/spin/issues/193)
-> so languages that do not have support for the component model can be used to
-> build Redis components for Spin.
+to build a component that Spin can invoke.
+
+This is exactly how Spin SDKs such as the [Rust](rust-components) and [Go](go-components) SDKs are built.
```

Comment:
```suggestion
This is exactly how Spin SDKs such as the [Rust](rust-components) and [Go](go-components) are built.
```

Reply:
Is there a word missing from the suggestion?  "...such as the Rust and Go ones..."?  I'm happier with saying SDKs; if we care deeply about avoiding the repetition then we could change one of them to "libraries" or something.

Reply:
@itowlson just felt like a little stutter. Perhaps a comma would help?

`This is exactly how Spin SDKs, such as the [Rust](rust-components) and [Go](go-components) SDKs, are built.`

OR 

`This is exactly how Spin SDKs are built such as the [Rust](rust-components) and [Go](go-components) SDKs.`

However, free to go with the original; i have sense i'm being a pedant.

Reply:
I like the commas idea - done and thanks!